vcd/datasource_vcd_nsxv_firewall.go

```diff
@@ -92,6 +92,15 @@ func datasourceVcdNsxvFirewallRule() *schema.Resource {
 							},
 						},
 						"virtual_machine_ids": {
+							Type:        schema.TypeSet,
+							Computed:    true,
+							Deprecated:  "In favor of vm_ids",
```

Comment:
Shouldn't I get a warning when using this parameter:

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # vcd_nsxv_firewall_rule.demo_nsxv_fw_rule will be updated in-place
  ~ resource "vcd_nsxv_firewall_rule" "demo_nsxv_fw_rule" {
        action          = "accept"
        edge_gateway    = "demo-gw"
        enabled         = true
        id              = "131078"
        logging_enabled = false
        name            = "demo-allow-https"
        rule_tag        = 131078
        rule_type       = "user"

      ~ destination {
            exclude             = false
            gateway_interfaces  = []
            ip_addresses        = [
                "10.150.160.116",
            ]
            ip_sets             = []
            org_networks        = []
          ~ virtual_machine_ids = [
              + "urn:vcloud:vm:b11ec7ce-3953-4bcc-a9a8-25e1ccb4c204",
            ]
            vm_ids              = []
        }

        service {
            port        = "4443"
            protocol    = "tcp"
            source_port = "any"
        }

        source {
            exclude             = false
            gateway_interfaces  = []
            ip_addresses        = [
                "any",
            ]
            ip_sets             = []
            org_networks        = []
            virtual_machine_ids = []
            vm_ids              = []
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

vcd_nsxv_firewall_rule.demo_nsxv_fw_rule: Modifying... [id=131078]
vcd_nsxv_firewall_rule.demo_nsxv_fw_rule: Still modifying... [id=131078, 10s elapsed]
vcd_nsxv_firewall_rule.demo_nsxv_fw_rule: Modifications complete after 19s [id=131078]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```

Reply:
Seems terraform don't see deprecations in TypeList.

Reply:
Let's then print the warning, because otherwise users won't know that they need to switch to vm_ids.

Reply:
Add custom warning, only visible on "apply" stage

Reply:
After testing, it's clear that there's no clean way to keep the old `virtual_machine_ids` field as deprecated in this case. When you try to clear the value apply ends up cycling on this:
```
  # vcd_nsxv_firewall_rule.demo_nsxv_fw_rule will be updated in-place
  ~ resource "vcd_nsxv_firewall_rule" "demo_nsxv_fw_rule" {
        action          = "accept"
        edge_gateway    = "demo-gw"
        enabled         = true
        id              = "131078"
        logging_enabled = false
        name            = "demo-allow-https"
        rule_tag        = 131078
        rule_type       = "user"
      ~ destination {
            exclude             = false
            gateway_interfaces  = []
            ip_addresses        = [
                "10.150.160.116",
            ]
            ip_sets             = []
            org_networks        = []
          ~ virtual_machine_ids = [
              - "urn:vcloud:vm:b11ec7ce-3953-4bcc-a9a8-25e1ccb4c204",
            ]
            vm_ids              = [
                "urn:vcloud:vm:b11ec7ce-3953-4bcc-a9a8-25e1ccb4c204",
            ]
        }
```

Reply:
Confirmed that the new approach of simply renaming from `virtual_machine_ids` to `vm_ids` works as expected in `vcd_nsxv_firewall_rule`.